smack-jingle-old/src/main/java/org/jivesoftware/smackx/jingleold/packet/Jingle.java

```diff
@@ -354,7 +354,7 @@ protected IQChildElementXmlStringBuilder getIQChildElementBuilder(IQChildElement
             buf.append(" responder=\"").append(getResponder()).append('"');
         }
         if (getAction() != null) {
-            buf.append(" action=\"").append(getAction().name()).append('"');
+            buf.append(" action=\"").append(getAction().toString()).append('"');
```

Comment:
I'm missing an explanation regarding what is wrong about using `name()` here?

Reply:
I'd assume that since `JingleActionEnum` takes a String value in its constructor, `action.name()` would return for example `CONTENT_ACCEPT` which is the name of the enum, while `action.toString()` will return the value of actionCode (in this case `content-accept`), which is what you actually want to put into the XML.

Reply:
Yes, exactly. 
And because of that, the parser sets `action=null` due to `JingleActionEnum` `getActiongetAction(String inActionCode)` method.
See also this discussion: https://discourse.igniterealtime.org/t/null-pointer-exception-in-jinglemanager/73226/5